magpie/api/management/group/group_views.py

```diff
@@ -124,48 +117,31 @@ def delete_group_view(request):
 @view_config(route_name=s.GroupUsersAPI.name, request_method="GET")
 def get_group_users_view(request):
     """
-    List all users from a group. Any pending user is not included in this list.
+    List all users from a group.
+    Users can be of active or pending type, or include both types depending of input arguments.
     """
     group = ar.get_group_matchdict_checked(request)
-    user_names = ax.evaluate_call(lambda: [user.user_name for user in group.users],
-                                  http_error=HTTPForbidden,
-                                  msg_on_fail=s.GroupUsers_GET_ForbiddenResponseSchema.description)
-    return ax.valid_http(http_success=HTTPOk, detail=s.GroupUsers_GET_OkResponseSchema.description,
-                         content={"user_names": sorted(user_names)})
-
-
-@s.GroupPendingUsersAPI.get(schema=s.GroupPendingUsers_GET_RequestSchema, tags=[s.GroupsTag],
-                            response_schemas=s.GroupPendingUsers_GET_responses)
-@view_config(route_name=s.GroupPendingUsersAPI.name, request_method="GET")
-def get_group_pending_users_view(request):
-    """
-    List all pending users from a group.
-    Pending users are users who requested to join a group but still need to accept terms and conditions.
-    """
-    # Find the group associated with the request
-    group_name = ar.get_value_matchdict_checked(request, "group_name")
-    group = ax.evaluate_call(lambda: GroupService.by_group_name(group_name, db_session=request.db),
-                             fallback=lambda: request.db.rollback(), http_error=HTTPForbidden,
-                             msg_on_fail=s.GroupPendingUsers_GET_ForbiddenResponseSchema.description)
-    ax.verify_param(group, not_none=True, http_error=HTTPNotFound,
-                    msg_on_fail=s.Group_MatchDictCheck_NotFoundResponseSchema.description)
+    user_type = ar.get_multiformat_body(request, "user_type", default=UserGroupType.ACTIVE_USERGROUPS.value)
 
-    # Find all temporary tokens with requested group id that have a pending accept terms request
-    query = request.db.query(TemporaryToken).filter(TemporaryToken.group_id == group.id)
-    tmp_tokens = query.filter(TemporaryToken.operation == TokenOperation.GROUP_ACCEPT_TERMS)
-
-    # Find and return all user names associated with the discovered tokens
-    pending_user_names = [tmp_token.user.user_name for tmp_token in tmp_tokens]
-
-    # Remove any user already belonging to the group, in case any tokens are irrelevant.
-    # Should not happen since related tokens are deleted upon T&C acceptation.
+    user_names = []
     member_user_names = ax.evaluate_call(lambda: [user.user_name for user in group.users],
                                          http_error=HTTPForbidden,
                                          msg_on_fail=s.GroupUsers_GET_ForbiddenResponseSchema.description)
```

Comment:
skip unnecessary fetch requested is pending only

Reply:
I currently use the member_user_names variable even for pending only.
I could remove it if it is too much, but I was removing any member names found in the temporary tokens, in case any tokens was remaining for an existing member. It is not supposed to happen, but if any token would remain, it would indicate the user as pending, even if he was really a member, since we rely on the tokens to find pending users.

Technically, I do remove any tokens, when the T&C are accepted, but I was doing this in case.
Let me know if you'd rather I remove this step.

Reply:
Ok. I see. Then it's fine to leave it like this.